src/main/java/io/outfoxx/swiftpoet/CodeWriter.kt

```diff
@@ -406,10 +408,59 @@ internal class CodeWriter constructor(
     }
   }
 
+  override fun close() {
+    out.close()
+  }
+
   /**
    * Returns the modules that should have been imported for this code.
    */
-  fun suggestedImports(): Map<String, DeclaredTypeName> {
+  private fun suggestedImports(): Map<String, DeclaredTypeName> {
     return importableTypes
   }
+
+  companion object {
+    /**
+     * Makes a pass to collect imports by executing [emitStep], and returns an instance of
+     * [CodeWriter] pre-initialized with collected imports.
+     */
+    fun withCollectedImports(
```

Comment:
Is this to replace `suggestedImports` or can we make this internal/private?

Reply:
no I do not think it replaces `suggestedImports`. Its just a different way to load imports since the file generation happens in two passes, once to figure out imports and then pass them into a new CodeWriter instance that actually emits them.

Reply:
Do we want it to be public API?

Reply:
I do not think so tbh but FileSpec refers to it.